File: 2022/py/day10.py
with open("input/day10.txt", "r") as f:
    data = f.read().splitlines()
    ans = 0
    cycle = 1
    x = 1
    pix = ""
    for line in data:
        if cycle in (20, 60, 100, 140, 180, 220): ans += x * cycle
        if (cycle-1) % 40 == 0: pix += '\n'
        if (cycle-1) % 40 in (x-1, x, x+1): pix += '#'
        else: pix += '.'
        if line.startswith("noop"):
            cycle += 1
        else:
            if cycle % 40 == 0: pix += '\n'
            if cycle % 40 in (x-1, x, x+1): pix += '#'
            else: pix += '.'
            if (cycle+1) in (20, 60, 100, 140, 180, 220): ans += x * (cycle+1)
            cycle += 2
            x += int(line.split()[1])

print(ans)
print(pix)

File: 2022/cpp/day10.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>
#include <vector>
using namespace std;

int check[] = {20, 60, 100, 140, 180, 220};

int main() {
    ifstream infile;
    infile.open("input/day10.txt", ios::in);
    string cur, ins, line = "";
    int val, cycle = 1, x = 1, i = 0, ans = 0;
    while (getline(infile, cur)) {
        stringstream ss(cur);
        ss >> ins >> val;
        if ((cycle-1) % 40 == 0) line += '\n';
        if ((cycle-1) % 40 == x || (cycle-1) % 40 == x+1 || (cycle-1) % 40 == x-1) line += '#';
        else line += '.';
        if (cycle == check[i] || (cycle == check[i]-1 && ins == "addx")) ans += x * check[i++];
        if (ins == "noop") cycle++;
        else {
            if (cycle % 40 == 0) line += '\n';
            if (cycle % 40 == x || cycle % 40 == x+1 || cycle % 40 == x-1) line += '#';
            else line += '.';
            x += val;
            cycle += 2;
        }
    }
    cout << ans << '\n';
    cout << line << '\n';
    return 0;
}